pkg/services/contexthandler/contexthandler.go

```diff
@@ -139,8 +140,7 @@ func (h *ContextHandler) Middleware(next http.Handler) http.Handler {
 
 		// End the span to make next handlers not wrapped within middleware span
 		span.End()
-
-		next.ServeHTTP(w, r.WithContext(identity.WithRequester(ctx, id)))
```

Comment:
`id` is nil here in some cases. The best place to extend the context is up above in the if-else. Moved to the correct block there.